### DISCUSSION TITLE
Страницы для "чужих" пространств идентификаторов

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#253] [Минимальный набор артефактов]
Бывают пространства идентификаторов, которые созданы не в рамках RuCore, например номер свидетельства о рождении, номер паспорта.
Есть следующие варианты присвоения идентификатора NamingSystem.uniqueId.value самим этим пространствам.

**Вариант 1. Создать свое описание пространства**
Придумать url, например
```
Instance:   core-ns-rf-birth-certificate
InstanceOf: NamingSystem
* name =   "Core_Ns_Rf_Birth_Certificate"

* uniqueId
  * type = #uri
  * value = "http://fhir.ru/core/systems/birth-certificate"
```
В этом случае можно на нашем сайте сделать страничку, куда будет вести этот путь, и на страничке давать информацию о системе именования, например нашу оценку надежности идентификатора, и возможно ссылки на внешние ресурсы.

**Вариант 2. Не создавать свое описание пространства.**

**Comment by Vrodvi** [comment]
Как обычно, посмотрел "лучшие практики" от HL7 - какие внешние пространства объявлены в базовой спецификации.
У всех просмотренных описаний внешних пространств идентификаторов, у которых в качестве предпочтительного uniqueId объявлен URL путь, этот URL путь никуда не ведёт, а объявлен он чисто как незначащий абстрактный идентификатор.

Вот примеры:
https://terminology.hl7.org/NamingSystem-passportNumNS-RUS.html
<img width="144" alt="image" src="https://github.com/fhir-ru/core/assets/92928137/9fcac00e-e6a5-4156-8bf1-01798fc8ac85">
https://terminology.hl7.org/NamingSystem-imei-identifier.html
https://terminology.hl7.org/NamingSystem-bluetooth-address-identifier.html		
https://terminology.hl7.org/NamingSystem-ssn.html	
	
Иногда попадаются рекомендации пойти посмотреть что-то по внешним ссылкам, но делается это просто в виде ссылки в текстовке самой NamingSystem, к примеру дана ссылка на базовый сайт dicom:
https://terminology.hl7.org/NamingSystem-dui.html

Поэтому предлагаю не создавать дополнительных страниц (информации для их заполнения у нас всё равно не наскребётся), оставить предпочтительные идентификаторы незначащими URL'ами, которые никуда не ведут, а всю необходимую (известную нам) информацию о пространстве идентификаторов просто указывать в описательной части NamingSystem (в атрибуте description, тип данных там markdown, так что можно ещё и красиво оформить это описание, при необходимости).

Пример:
```
* description = "An OID issued under DICOM OID rules. DICOM OIDs are represented as plain OIDs, with a prefix of \"urn:oid:\". See https://www.dicomstandard.org/"
```

```
"description" : "An OID issued under DICOM OID rules. DICOM OIDs are represented as plain OIDs, with a prefix of \"urn:oid:\". See https://www.dicomstandard.org/",
```

**Reply by eugenekogan-spb** [reply]
@OlegPenzin

**Comment by eugenekogan-spb** [comment]
Дополнительно к обсуждению 12.12.2023, где мы приняли решение оставить URL, которые никуда не ведут:
сайт руководства https://rucore.fhir.ru из актуальных текстов собран, но с ошибками:
<img width="1216" alt="image" src="https://github.com/fhir-ru/core/assets/62525329/9aaf8788-fc43-409b-bf67-1505c7e1b0c1">

Unable to provide support for code system http://fhir.ru/core/CodeSystem/core-cs-nsi-identity-document

Не в том ли проблема, что эта ссылка никуда не ведет?
@Vrodvi

**Reply by Vrodvi** [reply]
Моё предположение, что это из-за установленного типа наполнения - "пример" (` "content" : "example",`).
Поскольку наполнение справочника представлено в ресурсе не в полном объёме и явно указано, что представленное наполнение - это только пример, то инструмент не может делать полноценной проверки использованных в экземплярах кодов на соответствие этого фрагментарному наполнению.

**Для полноценной проверки мне нужно получить полный проекта SUSHI, который должен в автоматическом режиме собираться из Zendoc - где и как я могу получить такой проект и возможно ли это? Если нет, то прошу передать Тимуру соответствующее задание - должна иметься возможность получения собранного проекта SUSHI для проведения собственного анализа в рамках подобных проблем.**



Пока могу обозначить "плохие" варианты решения данной проблемы - использовать вместо типа "пример" типы "фрагмент" или "полное наполнение" - скорее всего ошибка уйдёт, но эти типы будут прямо противоречить реальному наполнению ресурса, что будет вводить читателей в заблуждение.

Более правильным вариантом будет включение полного наполнения справочника и установка значения "полное наполнение", но в этом случае всё равно нет гарантий, что не выйдет новая версия и наполнение опять устареет. А следовательно придётся писать сноску, что наполнение полное на состояние от такого-то числа, когда версия справочника была такая-то (так в СЭМДах делают). Тоже не ахти какой вариант, но уже получше.

Ну или вспоминаем, что самураи обещали "подумать" над автоматической загрузкой необходимых справочников в упрощённом виде (только код и  расшифровку кода), но зато чтобы точно все коды были загружены - ровно то что нужно для выполнения валидации.

**Reply by Vrodvi** [reply]
@eugenekogan-spb 
Ещё прошу передать Тимуру замечание по генерации стандартного руководства - при генерации добавляется очень много лишних, мусорных, данных и атрибутов - какие-то заглушки (placeholder), какой-то example.org в качестве издателя ресурса ("name" : "Example Publisher",).
Особенно "вредным является повсеместное указание версии 0.1.0 - она бездумно выставлена абсолютно всем ресурсам. Для справочников это особенно критично, т.к. вместо реально существующей версии справочника с ФНСИ, к примеру, читатель видит непонятно откуда взявшуюся и что означающую версию 0.1.0 - **это нужно обязательно убрать**.

![image](https://github.com/fhir-ru/core/assets/92928137/a2b326fe-9e91-46a4-bc38-a025407acd3e)
![image](https://github.com/fhir-ru/core/assets/92928137/5760ae8f-1ae2-4154-a4fe-a79a1c98cd14)

**Reply by eugenekogan-spb** [reply]
Ждем Тимура
